### PR TITLE
Feature/load albums from backend, edit ocelot api ports and routes

### PR DESCRIPTION
**Pull request by Den313**
Implemented loading albums (new-releases) from database. Also edited Ocelot API routing.

### CODE REVIEW

File: frontend/src/app/moduls/main/main-home/main-home.component.ts [hunk 1]
```diff
@@ -48,7 +55,16 @@ export class MainHomeComponent implements OnInit {
...
+      .pipe(takeUntil(this._unsubscribe$))
+      .subscribe(
+        (resp: HttpResponse<AlbumView[]>) => {
+          this.newReleases = resp.body!;
+        }
+      );
+  }
 
   // User should be able to play Calm rhythms - the newest playlists which moderator creates
   getCalmRhythms = (): Array<object> => new Array<object>();
```
Comment (proskurnicha) [inline frontend/src/app/moduls/main/main-home/main-home.component.ts]: Using takeUntil(this._unsubscribe$)
But ngOnDestroy not implemented

File: backend/Perflow/Services/AlbumsService.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            }
+            catch (ArgumentNullException ex)
+            {
+                throw new ArgumentNullException(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
+        }
```
Comment (proskurnicha) [inline backend/Perflow/Services/AlbumsService.cs]: We've already had a global exception handler, is it works?

File: backend/Perflow/Startup.cs [hunk 3]
```diff
@@ -49,7 +49,12 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 app.UseDeveloperExceptionPage();
                 app.UseSwagger();
                 app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "Perflow v1"));
-            }
+            }
+            app.UseCors(builder => builder
+                .AllowAnyMethod()
+                .AllowAnyHeader()
+                .AllowCredentials()
+                .WithOrigins("http://localhost:4200"));
```
Comment (proskurnicha) [inline backend/Perflow/Startup.cs]: Transfer hardcoded "http://localhost:4200" to appsettings.json

File: backend/Perflow/Services/AlbumsService.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        public async Task<Album> AddEntityAsync(Album albumDTO)
+        {
+            if (albumDTO == null)
+                throw new ArgumentNullException("Argument cannot be null");
+
+            try
+            {
+                albumDTO.Id = 0;
```
Comment (proskurnicha) [inline backend/Perflow/Services/AlbumsService.cs]: Why do we need it?

**Review by proskurnicha — CHANGES_REQUESTED** [review]

File: backend/Perflow/Common/DTO/Album/AlbumViewDTO.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Perflow.Common.DTO.Album
+{
+    public class AlbumViewDTO
+    {
+        public string Name { get; set; }
+        public string IconURL { get; set; }
+        public bool isSingle { get; set; }
```
Comment (NikitaKrasnovSW) [inline backend/Perflow/Common/DTO/Album/AlbumViewDTO.cs]: Property names should use PascalCase

**Review by NikitaKrasnovSW — APPROVED** [review]

**Review by NikitaKrasnovSW — CHANGES_REQUESTED** [review]
Needs fixes

**Review by NikitaKrasnovSW — APPROVED** [review]

**Pull request merged**